adobe/react-spectrum
# [SSR] Removed call of requestAnimationFrame outside of function

This allows the base library of @adobe/react-spectrum to used in Gatsby

Closes #842

## ✅ Pull Request Checklist:


## 📝 Test Instructions:
- Create a basic hello-world gatsby app
- Add this snippet to your app
```javascript
<Provider theme={defaultTheme}>
  <Flex direction="column" width="size-2000" gap="size-100">
      <View>this is a thing</View>
      <View>this is a thing</View>
      <View>this is a thing</View>
      <View>this is a thing</View>
    </Flex>
</Provider>
```
- Build!

Without this change, the gatsby build will try to run rAF as it is outside of a actually hook function that would normally be called in the render loop. This caused the error in #835. With this simple change, functionality is essentially the same, but it will now compile in a Gatsby enviroment.
The only real loss is that the check will happen every animation frame, but it is a semi-cheap comparison so it shouldn't effect any actual animations.
See https://developers.google.com/web/updates/2012/05/requestAnimationFrame-API-now-with-sub-millisecond-precision#feature_detection to show the change shouldn't make any reasonable difference to the code.
